### PR TITLE
Add Giotto-tda

### PR DESCRIPTION
**Pull request by giotto-learn**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by gtauzin** [comment]
Hi,

The build fails because a few C++ files cannot be found. giotto-tda has C++ dependency included in the form of git submodules, which AFAIK will not be in the git tarball. We use GitHub tags and PyPI for publishing new releases. Is there a way we could make that work?

Thank you in advance for your support!

File: recipes/giotto-tda/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    - python
+    - pybind11
+    - boost >=1.56.*
+    - setuptools
+    - pip
+    - numpy >=1.17.0
+    - scipy >=0.17.0
+    - joblib >=0.13
+    - scikit-learn >=0.22.0
+    - python-igraph >=0.7.1.post6
```
Comment (scopatz) [inline recipes/giotto-tda/meta.yaml:33]: Please remove the post part of these deps

**Comment by scopatz** [comment]
You are proabably not setting `$PREFIX` correctly

File: recipes/giotto-tda/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  version: {{ version }}
+
+source:
+  url: "https://github.com/giotto-ai/{{ name }}/archive/v{{ version }}.tar.gz"
+  sha256: 36e637a4cc9beb9454932534dbc790f0871698230546737e3eba101dd3ac1b4e
+
+build:
+  number: 0
+  skip: true  # [py2k]
+  skip: true  # [vs<17]
```
Comment (scopatz) [inline recipes/giotto-tda/meta.yaml:15]: Please just have one skip line with `[py<3 or vs<17]`

File: recipes/giotto-tda/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    - setuptools
+    - pip
+    - numpy >=1.17.0
+    - scipy
+    - joblib
+    - scikit-learn
+    - python-igraph
+    - plotly >=4.4.1
+    - ipywidgets >=7.5.1
+    - openmp
```
Comment (scopatz) [inline recipes/giotto-tda/meta.yaml:35]: Are all of these needed at build time?

**Comment by scopatz** [comment]
Thanks!

File: recipes/giotto-tda/meta.yaml (hunk 1 above)
Comment (ulupo) [reply recipes/giotto-tda/meta.yaml]: @scopatz FYI, this dependency has now been bumped to `0.8.2`.

**Pull request merged**